### PR TITLE
EVA-2464 Correct contig error in study PRJEB43246

### PR DESCRIPTION
**Pull request by nitin-ebi**

### CODE REVIEW

File: tasks/eva_2461/correct_contig_error_in_study.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    keys = ['seq', 'study', 'contig', 'start', 'ref', 'alt']
+    h.update('_'.join([str(variant_rec[key]) for key in keys]).encode())
+    return h.hexdigest().upper()
+
+
+def replace_with_correct_contig(mongo_source):
+    correct_contig = 'AF034253.1'
+    sve_collection = mongo_source.mongo_handle[mongo_source.db_name]["submittedVariantEntity"]
+    filter_criteria = {'seq': 'GCA_000003025.4', 'study': 'PRJEB43246', 'contig': 'M'}
+    cursor = sve_collection.find(filter_criteria)
```
Comment (sundarvenkata-EBI) [inline tasks/eva_2461/correct_contig_error_in_study.py:22]: Even though we know the number of documents is small,  Mongo is trying to find a needle in a haystack, I would suggest providing a parameter for cursor timeout with a generous value here.

**Comment by nitin-ebi** [comment]
Code adapted from existing script [correct_contig_with_chr_removed](https://github.com/EBIvariation/eva-tasks/blob/master/tasks/eva_2287/correct_contig_with_chr_removed.py)

File: tasks/eva_2461/correct_contig_error_in_study.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    try:
+        for variant in cursor:
+            original_id = get_SHA1(variant)
+            assert variant['_id'] == original_id, "Original id is different from the one calculated %s != %s" % (
+                variant['_id'], original_id)
+            variant['contig'] = correct_contig
+            variant['_id'] = get_SHA1(variant)
+            insert_statements.append(pymongo.InsertOne(variant))
+            drop_statements.append(pymongo.DeleteOne({'_id': original_id}))
+        result_insert = sve_collection.bulk_write(requests=insert_statements, ordered=False)
```
Comment (sundarvenkata-EBI) [inline tasks/eva_2461/correct_contig_error_in_study.py:36]: Suggest using majority write concern here...

File: tasks/eva_2461/correct_contig_error_in_study.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        total_inserted += result_insert.inserted_count
+        result_drop = sve_collection.bulk_write(requests=drop_statements, ordered=False)
```
Comment (sundarvenkata-EBI) [inline tasks/eva_2461/correct_contig_error_in_study.py:38]: Suggest using majority write concern here...

**Review by sundarvenkata-EBI — CHANGES_REQUESTED** [review]
Looks good! Left a few comments regarding read/write concerns and timeouts.

File: tasks/eva_2461/correct_contig_error_in_study.py (hunk 1 above)
Comment (tcezard) [reply tasks/eva_2461/correct_contig_error_in_study.py:22]: I would advise to pass `no_cursor_timeout=True` see [doc](https://pymongo.readthedocs.io/en/stable/api/pymongo/collection.html#pymongo.collection.Collection.find)

**Review by tcezard — APPROVED** [review]

File: tasks/eva_2461/correct_contig_error_in_study.py (hunk 1 above)
Comment (nitin-ebi) [reply tasks/eva_2461/correct_contig_error_in_study.py]: updated

File: tasks/eva_2461/correct_contig_error_in_study.py (hunk 3 above)
Comment (nitin-ebi) [reply tasks/eva_2461/correct_contig_error_in_study.py]: updated

File: tasks/eva_2461/correct_contig_error_in_study.py (hunk 2 above)
Comment (nitin-ebi) [reply tasks/eva_2461/correct_contig_error_in_study.py]: updated

**Review by sundarvenkata-EBI — APPROVED** [review]

File: tasks/eva_2461/correct_contig_error_in_study.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    keys = ['seq', 'study', 'contig', 'start', 'ref', 'alt']
+    h.update('_'.join([str(variant_rec[key]) for key in keys]).encode())
+    return h.hexdigest().upper()
+
+
+def replace_with_correct_contig(mongo_source):
+    correct_contig = 'AF034253.1'
+    sve_collection = mongo_source.mongo_handle[mongo_source.db_name]["submittedVariantEntity"]
+    filter_criteria = {'seq': 'GCA_000003025.4', 'study': 'PRJEB43246', 'contig': 'M'}
+    cursor = sve_collection.find(filter_criteria, no_cursor_timeout=True)
```
Comment (sundarvenkata-EBI) [inline tasks/eva_2461/correct_contig_error_in_study.py:23]: Could you also check to see if you can use "majority" readConcern here?

Comment (nitin-ebi) [reply tasks/eva_2461/correct_contig_error_in_study.py]: updated

**Review by sundarvenkata-EBI — APPROVED** [review]

**Pull request merged**